Sources/Logging/Docs.docc/ImplementingALogHandler.md

```diff
@@ -0,0 +1,171 @@
+some global log level override that may affect all log handlers created.
+
+## Example Implementation
+
+Here's a complete example of a simple print-based log handler:
+
+```swift
+import Foundation
+import Logging
+
+public struct PrintLogHandler: LogHandler {
+    private let label: String
+    public var logLevel: Logger.Level = .info
+    public var metadata: Logger.Metadata = [:]
+    
+    public init(label: String) {
+        self.label = label
+    }
+    
+    public func log(
+        level: Logger.Level,
+        message: Logger.Message,
+        metadata: Logger.Metadata?,
+        source: String,
+        file: String,
+        function: String,
+        line: UInt
+    ) {
+        let timestamp = ISO8601DateFormatter().string(from: Date())
+        let levelString = level.rawValue.uppercased()
+        
+        // Merge handler metadata with message metadata
+        let combinedMetadata = self.metadata
+            .merging(metadata ?? [:]) { _, new in new }
+        
+        // Format metadata
+        let metadataString = combinedMetadata.map { "\($0.key)=\($0.value)" }.joined(separator: ",")
+        
+        // Create log line and print to console
+        let logLine = "\(label) \(timestamp) \(levelString) [\(metadataString)]: \(message)"
+        print(logLine)
+    }
+    
+    public subscript(metadataKey key: String) -> Logger.Metadata.Value? {
+        get {
+            return self.metadata[key]
+        }
+        set {
+            self.metadata[key] = newValue
+        }
+    }
+}
+
+```
+
+### Advanced Features
+
+#### Metadata Providers
+
+Metadata providers allow you to dynamically add contextual information to all
+log messages without explicitly passing it each time. Common use cases include
+request IDs, user sessions, or trace contexts that should be included in logs
+throughout a request's lifecycle.
+
+Support dynamic metadata injection:
+
+```swift
+import Foundation
+import Logging
+
+public struct PrintLogHandler: LogHandler {
+    private let label: String
+    public var logLevel: Logger.Level = .info
+    public var metadata: Logger.Metadata = [:]
+    public var metadataProvider: Logger.MetadataProvider?
+    
+    public init(label: String) {
+        self.label = label
+    }
+    
+    public func log(
+        level: Logger.Level,
+        message: Logger.Message,
+        metadata: Logger.Metadata?,
+        source: String,
+        file: String,
+        function: String,
+        line: UInt
+    ) {
+        let timestamp = ISO8601DateFormatter().string(from: Date())
+        let levelString = level.rawValue.uppercased()
+        
+        // Get provider metadata
+        let providerMetadata = metadataProvider?.get() ?? [:]
+        
+        // Merge all metadata sources
+        let combinedMetadata = providerMetadata
+            .merging(self.metadata) { _, new in new }
+            .merging(metadata ?? [:]) { _, new in new }
```

Comment:
heh I wonder if this should show a more efficient way if we're going to show people a copy-paste-ready snippet?

Our own one does:  https://github.com/apple/swift-log/blob/main/Sources/Logging/Logging.swift#L1404